### PR TITLE
Nothing else should be interactive when a dropdown is open

### PR DESCRIPTION
**Pull request by jesusbotella**
Related to #13132 
----

This PR adds an overlay below dropdowns to disable interaction on elements behind it. When any dropdown is open, it should disable all interactions between elements including activations and hover.

This PR includes an associated PR in CartoAssets: https://github.com/CartoDB/CartoAssets/pull/186

### Acceptance
1. Go to any of your created maps
2. Check that interaction is disabled when three points button is clicked
    2.1. Click on any of the three points buttons to open the dropdown and check that interaction between elements is disabled. You can find them all over the Builder, but the easiest ones can be found on the map sidebar header, the layers list, and the widgets list.
3. Check that interaction is disabled when color picker is open
    3.1. Go to any layer on your map
    3.2. Click on the style tab
    3.3. Click on the color picker
    3.4. Check that interaction is disabled between elements
4. Check that interaction is disabled when selectable list dropdown is open
    4.1. Go to any layer on your map
    4.2. Click on the style tab
    4.3. Click on the blending input
    4.4. Check that interaction is disabled between elements

### CODE REVIEW

**Comment by matallo** [comment]
updating NEWS and dependencies myself

File: lib/assets/core/javascripts/cartodb3/components/context-menu/context-menu-view.js [hunk 1]
```diff
@@ -90,6 +88,11 @@ module.exports = CoreView.extend({
       })
     );
 
+    this.dropdownOverlay = new DropdownOverlayView();
+    this.dropdownOverlay.toggle(this.isVisible());
+    this.dropdownOverlay.bind('overlayClicked', this.hide, this);
```
Comment (matallo) [inline lib/assets/core/javascripts/cartodb3/components/context-menu/context-menu-view.js]: this looks like a weird pattern to me, why not passing the `this.hide()` as a parameter, like `onClickAction: this.hide()` and handle the bond or listener at dropdownOverlay view?

File: lib/assets/core/javascripts/cartodb3/components/context-menu/context-menu-view.js [hunk 2]
```diff
@@ -90,6 +88,11 @@ module.exports = CoreView.extend({
       })
     );
 
+    this.dropdownOverlay = new DropdownOverlayView();
+    this.dropdownOverlay.toggle(this.isVisible());
```
Comment (matallo) [inline lib/assets/core/javascripts/cartodb3/components/context-menu/context-menu-view.js]: something similar to below, I'd handle this at the view initialize, and pass the visible as parameter

File: lib/assets/core/javascripts/cartodb3/components/dropdown-overlay/dropdown-overlay-view.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  show: function () {
+    this.$el.toggle(true);
+  },
+
+  hide: function () {
+    this.$el.toggle(false);
+  },
+
+  toggle: function (visibleState) {
+    this.$el.toggle(visibleState);
```
Comment (matallo) [inline lib/assets/core/javascripts/cartodb3/components/dropdown-overlay/dropdown-overlay-view.js]: I have a hard time trying to understand state management for visibility via this parameter, why not having a simple model with `visibility: false` attribute, then setting this attribute to true or false, and listen to change in that model? we're currently doing this several times in the app

File: lib/assets/core/javascripts/cartodb3/components/dropdown-overlay/dropdown-overlay-view.js [hunk 4]
```diff
@@ -14,38 +15,49 @@ module.exports = CoreView.extend({
   },
 
   initialize: function (opts) {
-    this.dropdownElement = opts && opts.dropdownElement;
+    this.container = opts && opts.container;
+    this.onClickAction = opts && opts.onClickAction;
+
+    this.model = new Backbone.Model({
+      visible: opts && typeof opts.visible !== 'undefined' ? opts.visible : true
```
Comment (matallo) [inline lib/assets/core/javascripts/cartodb3/components/dropdown-overlay/dropdown-overlay-view.js]: `_.isUndefined` or `opts.visible !== void 0`

Comment (jesusbotella) [reply lib/assets/core/javascripts/cartodb3/components/dropdown-overlay/dropdown-overlay-view.js]: I'll go with `_.isUndefined` :)

**Comment by rubenmoya** [comment]
It's not working for this dropdown, I think it's `Editor-dropdownOperators`:

select_click

**Comment by jesusbotella** [comment]
I forgot that one! Added overlay in 234d51ee88a0960997ffca2a05ab689f758a6526. Back to CR.

File: lib/assets/core/javascripts/cartodb3/components/form-components/editors/operators/operators-list-view.js [hunk 5]
```diff
@@ -38,6 +39,12 @@ module.exports = CoreView.extend({
       })
     );
     this._renderList();
+
+    this._dropdownOverlay = new DropdownOverlayView({
+      onClickAction: this.hide.bind(this)
+    });
+    this.addView(this._dropdownOverlay);
```
Comment (rubenmoya) [inline lib/assets/core/javascripts/cartodb3/components/form-components/editors/operators/operators-list-view.js]: Maybe add a test checking the subviews?

**Comment by rubenmoya** [comment]
Working as expected, great job! 💪 

Moving to deploy!

**Pull request merged**